bravado/testing/integration_test.py

```diff
@@ -412,15 +428,34 @@ def test_post_request(self, swagger_http_server):
 
         assert resp.text == self.encode_expected_response(b'6')
 
-    def test_boolean_header(self, swagger_http_server):
+    def test_non_string_headers(self, swagger_http_server):
+        headers = {
+            'Header-Boolean': True,
+            'Header-Integer': 1,
+            'Header-Bytes': b'0',
+        }
         response = self.http_client.request({
             'method': 'GET',
-            'headers': {'boolean-header': True},
-            'url': '{server_address}/1'.format(server_address=swagger_http_server),
+            'headers': headers,
+            'url': '{server_address}/headers'.format(server_address=swagger_http_server),
             'params': {},
         }).result(timeout=1)
 
-        assert response.text == self.encode_expected_response(ROUTE_1_RESPONSE)
+        expected_header_representations = {
+            header_name: repr(
+                str(header_value.decode())
+                if isinstance(header_value, bytes)
+                else str(header_value)
+            )
+            for header_name, header_value in headers.items()
+        }
```

Comment:
Please provide the actual expected headers; this will make the test more obvious to read and to understand, and it provides us (humans) with the ability to make sure things really are the way we want them to be. :)